llama_index/llms/base.py

```diff
@@ -337,3 +348,117 @@ async def astream_complete(
         self, prompt: str, **kwargs: Any
     ) -> CompletionResponseAsyncGen:
         """Async streaming completion endpoint for LLM."""
+
+    def _as_query_component(self, **kwargs: Any) -> QueryComponent:
+        """Return query component."""
+        if self.metadata.is_chat_model:
+            return LLMChatComponent(llm=self)
```

Comment:
Is it better to use `llm.predict()` here? 

It makes the decision between chat and completion for you, and also avoids outputting messages with the `assistant` prefix like what happens in the notebook

In the future, we can expand this to include streaming and structured predict modes too?

Reply:
predict takes a prompt template though, so i'm not actually sure how that would work? llm.predcit is basically its own small query pipeline

Reply:
I think taking a prompt template makes sense though? `predict()` also applies some key formatting like `llm.completion_to_prompt()` and `llm.messages_to_prompt()`

If the input is a string, we can convert it to a template on the fly

Reply:
I think using `predict()` makes sense, since it handles both formatting (messages_to_prompt and completion_to_prompt) as well as deciding chat vs completion (I'm just noticing a bug with this though, I will push a fix to this branch related to that)

We can cast to a template on the fly as well, if the input is a string 🤔

Reply:
or we can let the validate stuff handle calling the prompt formatting like it kind of does now

Reply:
(pushed a small-ish tweak to ensure that prompts should not get double-formatted)

Reply:
yeah that's true, we can just have it be a prompt template with 0 template args

Reply:
working on the change rn